code/game/objects/items/RCD.dm

```diff
@@ -1104,108 +1104,227 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
+	var/choice = show_radial_menu(user, src, options_menu, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	if(!check_menu(user))
 		return
 
-	//this is just cycling options through a list
-	var/current_loc = layers.Find(current_layer) + 1
+	switch(choice)
+		if("Current Layer")
+			choose_layer_menu(user)
+		if("Current Color")
+			choose_color_menu(user)
 
-	if(current_loc > layers.len)
-		current_loc = 1
+/**
+ * Choose the current layer via radial menu.
+ *
+ * Arguments:
+ * * user - current user.
+ */
+/obj/item/construction/plumbing/proc/choose_layer_menu(mob/user)
+	if(!GLOB.plumbing_layer_menu_options.len)
+		for(var/layer_name in GLOB.plumbing_layers)
+			GLOB.plumbing_layer_menu_options += list((layer_name) = image('icons/hud/radial.dmi', icon_state = "plumbing_layer[GLOB.plumbing_layers[layer_name]]"))
 
-	//We want the key (the define), not the index (the string)
-	current_layer = layers[current_loc]
-	to_chat(user, span_notice("You switch [src] to [current_layer]."))
+	playsound(src.loc, 'sound/effects/pop.ogg', 50, FALSE)
+	var/new_layer = show_radial_menu(user, src, GLOB.plumbing_layer_menu_options, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	if(!new_layer || !check_menu(user))
+		return
+
+	current_layer = new_layer
+	to_chat(user, span_notice("You set the layer to [new_layer]."))
+
+/**
+ * Choose the current color via radial menu.
+ *
+ * Arguments:
+ * * user - current user.
+ */
+/obj/item/construction/plumbing/proc/choose_color_menu(mob/user)
+	if(!GLOB.plumbing_color_menu_options.len)
+		for(var/color_name in GLOB.pipe_paint_colors)
+			GLOB.plumbing_color_menu_options += list((color_name) = image('icons/hud/radial.dmi', icon_state = color_name))
+
+	playsound(src.loc, 'sound/effects/pop.ogg', 50, FALSE)
+	var/new_color = show_radial_menu(user, src, GLOB.plumbing_color_menu_options, custom_check = CALLBACK(src, .proc/check_menu, user), require_near = TRUE, tooltips = TRUE)
+	if(!new_color || !check_menu(user))
+		return
+
+	current_color = new_color
+	to_chat(user, span_notice("You set the color to [new_color]."))
+
+/**
+ * Choose layer via mouse wheel, like an RPD
+ *
+ * Arguments:
+ * * source - the user
+ * * A - the atom being selected, unused.
+ * * delta_x - X scroll delta
+ * * delta_y - Y scroll delta
+ */
+/obj/item/construction/plumbing/proc/mouse_wheeled(mob/source, atom/A, delta_x, delta_y, params)
+	SIGNAL_HANDLER
+	if(source.incapacitated(IGNORE_RESTRAINTS|IGNORE_STASIS))
+		return
+
+	if(delta_y < 0)
+		var/current_loc = GLOB.plumbing_layers.Find(current_layer) + 1
+		if(current_loc > GLOB.plumbing_layers.len)
+			current_loc = 1
+		current_layer = GLOB.plumbing_layers[current_loc]
+	else if(delta_y > 0)
+		var/current_loc = GLOB.plumbing_layers.Find(current_layer) - 1
+		if(current_loc < 1)
+			current_loc = GLOB.plumbing_layers.len
+		current_layer = GLOB.plumbing_layers[current_loc]
+	else
+		return
```

Comment:
check this first for the early return before delta_y < and > checks. Aka, `if(delta_y == 0) return` then rest of the code